.github/actions/setup-dotnet/action.yml

```diff
@@ -23,6 +28,17 @@ runs:
         "DOTNET_CLI_TELEMETRY_OPTOUT=true" >> $Env:GITHUB_ENV
         $Env:DOTNET_CLI_TELEMETRY_OPTOUT = "true"
 
+    # For wildcard .NET versions, the General Availability ("ga") channel should be selected. For concrete versions,
+    # this shouldn't be set, because that'll cause a warning.
+    - name: Set dotnet-quality
+      id: set-dotnet-quality
+      shell: pwsh
+      run: |
+        if ('${{ inputs.dotnet-version }}' -match 'x')
+        {
+            Set-GitHubOutput "dotnet-quality" "ga"
```

Comment:
Since you've set up the scripts to be able to use `Set-GitHubOutput`, it would be better to update the rest of the steps to use it as well.
BTW you can just use singlequotes here (the strings are constant).

Reply:
The other steps don't write to the output, but `GITHUB_ENV`, and we don't have a script for that. But now I created it and changed the references.

Reply:
Oh, yeah, I mixed them up! But it's a nice improvement in any case.

Reply:
Yep!